### PR TITLE
[MU4] fix #8773, "change enharmonic spelling" commands not implemented

### LINKED ISSUE #8773: [MU4 Issue] Change enharmonic spelling commands not implemented
**Describe the bug**
The "change enharmonic spelling" command has not been ported into MU4.
Also "respell pitches" to reverse this is applying to the *whole* score rather than just the selected notes - this is as per MU3, but is inconsistent with other commands and expected behaviour.

**Expected behavior**
The "change enharmonic spelling" commands mapped to shortcut key "J" and "ctrl+J" for (changing just concert pitch view or transposing pitch view, depending what's been viewed currently) should work as per MU3.

Respell pitches should revert these for just the selected notes, rather than affecting other notes (unless no notes are selected, then it can safely apply to the whole score)

**Screenshots**

**Desktop (please complete the following information):**
Windows 10

**Additional context**

### PR DESCRIPTION
**Pull request by wizofaus**
Resolves: https://github.com/musescore/MuseScore/issues/8773

Ensures "change enharmonic spelling" commands available in MU3 are ported to MU4, and ensures "Respell pitches" only applies to selected notes

- [x] I signed [CLA](https://musescore.org/en/cla)
- [x] I made sure the code in the PR follows [the coding rules](https://github.com/musescore/MuseScore/wiki/CodeGuidelines)
- [x] I made sure the code compiles on my machine
- [x] I made sure there are no unnecessary changes in the code
- [x] I made sure the title of the PR reflects the core meaning of the issue you are solving
- [x] I made sure the commit message(s) contain a description and answer the question "Why do those changes fix that particular issue?" or "Why are those changes really necessary as improvements?"
- [x] I made sure the commit message title starts with "fix #424242:" if there is a related issue
- [ ] I created the test (mtest, vtest, script test) to verify the changes I made

### CODE REVIEW

**Comment by wizofaus** [comment]
If moving this functionality into score.cpp is considered appropriate, I'll add some unit tests for it (there were none in MU3 as it was in ScoreView.cpp).

File: src/engraving/libmscore/score.cpp [hunk 1]
```diff
@@ -880,6 +879,63 @@ void Score::spell(int startStaff, int endStaff, Segment* startSegment, Segment*
     }
 }
 
+void Score::changeEnharmonicSpelling(bool both)
+{
```
Comment (wizofaus) [inline src/engraving/libmscore/score.cpp:883]: copied from ScoreView.cpp in MU3

File: src/engraving/libmscore/score.cpp [hunk 2]
```diff
@@ -850,9 +850,8 @@ void Score::spell()
             for (int track = strack; track < etrack; ++track) {
                 Element* e = s->element(track);
                 if (e && e->type() == ElementType::CHORD) {
-                    notes.insert(notes.end(),
-                                 toChord(e)->notes().begin(),
-                                 toChord(e)->notes().end());
+                    std::copy_if(toChord(e)->notes().begin(), toChord(e)->notes().end(),
+                                 std::back_inserter(notes), [this](Element* ce) { return selection().isNone() || ce->selected(); });
```
Comment (Jojo-Schmitz) [inline src/engraving/libmscore/score.cpp:854]: This is the part that ensures "Respell pitches" only applies to selected notes, right?

Comment (Jojo-Schmitz) [reply src/engraving/libmscore/score.cpp:854]: And so fixes https://musescore.org/en/node/284064?

Comment (Jojo-Schmitz) [reply src/engraving/libmscore/score.cpp:854]: Apparently it is!

Comment (wizofaus) [reply src/engraving/libmscore/score.cpp:854]: Yes...

**Comment by bkunda** [comment]
Tested and approved from a design perspective. Thanks @wizofaus!

**Review by igorkorsukov — APPROVED** [review]
